### PR TITLE
Add analytics searchValue and searchId headers 

### PR DESCRIPTION
**Pull request by divyanshu013**
- Get rid of the componentTypes and corresponding reducer
- Adds a new reducer for analytics
- Adds a new action type for setting search id
- msearch method determines search query or search id per request
- Sets search filters header
- Adds utility methods for analytics
- Fixes tests and add unit tests for analytics

NOTE: This PR would also require the [corresponding PR](https://github.com/appbaseio/reactivesearch/pull/525) in reactivesearch to include component types in `updateQuery`.

### CODE REVIEW

File: src/actions/query.js [hunk 1]
```diff
@@ -81,32 +82,41 @@ export function setHeaders(headers) {
...
-		const validComponents = Object.keys(componentType)
-			.filter(item => SEARCH_COMPONENTS.includes(componentType[item]));
 
 		// send search id or term in headers
-		// TODO: implement support for search id
 		if (
 			config.analytics
 			&& config.url === 'https://scalr.api.appbase.io'
-			&& validComponents.length
+			&& !isInternalComponent
```
Comment (metagrover) [inline src/actions/query.js:110]: Why are we only filtering internal components here?

File: src/actions/query.js [hunk 2]
```diff
@@ -299,7 +314,8 @@ export function executeQuery(componentId, executeWatchList = false, mustExecuteM
 		});
 
 		if (finalQuery.length) {
-			dispatch(msearch(finalQuery, orderOfQueries));
+			// in case of an internal component the analytics headers should not be included
+			dispatch(msearch(finalQuery, orderOfQueries, false, componentId.endsWith('__internal')));
```
Comment (metagrover) [inline src/actions/query.js:327]: This is redundant because internal components never fire a query.

**Review by metagrover — CHANGES_REQUESTED** [review]

File: src/actions/query.js (hunk 1 above)
Comment (divyanshu013) [reply src/actions/query.js:110]: I'm using this condition to check if the `msearch` was invoked due to an internal component in which case I'm not adding the analytics headers.

File: src/actions/query.js (hunk 2 above)
Comment (divyanshu013) [reply src/actions/query.js:327]: This passes a parameter to the `msearch` function to tell if the query was executed due to an internal component. Related to above comment. If there's a better approach we could try it as well.

Comment (metagrover) [reply src/actions/query.js:327]: I think instead of blacklisting internal components, it would be best if we whitelist search analytics headers application only for the Result and ReactiveComponents.

File: src/actions/query.js (hunk 1 above)
Comment (metagrover) [reply src/actions/query.js:110]: Refer: https://github.com/appbaseio/reactivecore/pull/15#discussion_r218311514

File: src/actions/query.js (hunk 2 above)
Comment (divyanshu013) [reply src/actions/query.js:327]: But when filters change that should also apply the search headers. Currently its working with search filters as well. Updating the PR in a while for search-filters support. Lets review this condition then.

Comment (metagrover) [reply src/actions/query.js:327]: Resolving because we need search headers for filters as well

File: src/utils/__tests__/analytics.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	};
+	const res = parseFilterValue(componentId, componentValues);
+	expect(res).toBe('Filter=value1,Filter=value2');
+});
+
+test('parseFilterValue should parse array of object values', () => {
+	const componentId = 'id1';
+	const componentValues = {
+		label: 'Filter',
+		value: [{ label: 'label1', value: 'value1' }],
```
Comment (metagrover) [inline src/utils/__tests__/analytics.js:58]: You should also include the test for multiple values in the array and null values.

**Review by metagrover — CHANGES_REQUESTED** [review]

File: src/utils/__tests__/analytics.js (hunk 3 above)
Comment (divyanshu013) [reply src/utils/__tests__/analytics.js:58]: Added test for multiple values.

The `parseFilterValue` will return `null` as a filter value independently since it just parses the value. But a `null` value shouldn't get through which `getFilterString` is taking care of by filtering out components having `null` or falsy value.

Comment (metagrover) [reply src/utils/__tests__/analytics.js:58]: cool beans

**Review by metagrover — APPROVED** [review]
